pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -613,18 +613,12 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1.PipelineRun, getPipel
 
 		switch aaBehavior {
 		case affinityassistant.AffinityAssistantPerWorkspace, affinityassistant.AffinityAssistantDisabled:
-			if err = c.createOrUpdateAffinityAssistantsAndPVCs(ctx, pr, aaBehavior); err != nil {
+			failReason, err := c.createOrUpdateAffinityAssistantsAndPVCs(ctx, pr, aaBehavior)
```

Comment:
Rather than separating the error and the reason for the error, why not use sentinel errors? https://go.dev/blog/go1.13-errors

Here's an example: https://github.com/tektoncd/pipeline/blob/b769b5620300d7fb6d6638083124d03636caa503/pkg/reconciler/pipelinerun/pipelinerun.go#L364-L366